adam_core/propagator/propagator.py

```diff
@@ -127,29 +132,25 @@ def generate_ephemeris(
 
         Returns
         -------
-        ephemeris : (N * M)
+        ephemeris : List[`~quivr.linkage.MultiKeyLinkage`] (M)
             Predicted ephemerides for each orbit observed by each
             observer.
-
-        TODO: Add ephemeris class
-        TODO: Add an observers class
         """
         if max_processes is None or max_processes > 1:
-            with concurrent.futures.ProcessPoolExecutor(max_workers=max_processes) as executor:
+            with concurrent.futures.ProcessPoolExecutor(
+                max_workers=max_processes
+            ) as executor:
                 futures = []
                 for orbit_chunk in _iterate_chunks(orbits, chunk_size):
-                    futures.append(executor.submit(ephemeris_worker, orbit_chunk, observers, self))
+                    futures.append(
+                        executor.submit(ephemeris_worker, orbit_chunk, observers, self)
+                    )
 
                 ephemeris_list = []
                 for future in concurrent.futures.as_completed(futures):
                     ephemeris_list.append(future.result())
-            ephemeris = concatenate(ephemeris_list)
 
-        else:
-            ephemeris = self._generate_ephemeris(orbits, observers)
+            return ephemeris_list
 
-        ephemeris.sort_values(
-            by=["orbit_ids", "origin", "times"],
-            inplace=True,
-        )
-        return ephemeris
+        else:
+            return [self._generate_ephemeris(orbits, observers)]
```

Comment:
@spenczar Note that right now this function will return a list of MultiKeyLinkages in the absence of having a way to concatenate the linkages.

Reply:
With #33 merged, I think you could merge the linkages now.